### PR TITLE
storage/*: implement metrics for scan details in scheduler

### PR DESCRIPTION
**Pull request by AndreMouche**
@overvenus @disksing PTAL

### CODE REVIEW

File: src/storage/txn/scheduler.rs [hunk 1]
```diff
@@ -958,6 +963,33 @@ fn process_write_impl(
...
+        entry.add_statistics(stat);
+    }
+}
+
+impl ThreadContext for ScheContext {
+    fn on_tick(&mut self) {
+        for (cmd, stat) in &self.stats {
+            for (cf, details) in stat.stat.details() {
+                for (tag, count) in details {
+                    KV_COMMAND_SCAN_DETAILS
```
Comment (siddontang) [inline src/storage/txn/scheduler.rs]: too many `for` here 😓

File: src/storage/txn/scheduler.rs [hunk 2]
```diff
@@ -958,6 +963,33 @@ fn process_write_impl(
...
+impl ScheContext {
+    fn add_statistics(&mut self, cmd_tag: &'static str, stat: &Statistics) {
+        let entry = self.stats.entry(cmd_tag).or_insert_with(Default::default);
+        entry.add_statistics(stat);
+    }
+}
+
+impl ThreadContext for ScheContext {
+    fn on_tick(&mut self) {
+        for (cmd, stat) in &self.stats {
```
Comment (siddontang) [inline src/storage/txn/scheduler.rs]: you can use drain here

File: src/storage/txn/scheduler.rs [hunk 3]
```diff
@@ -958,6 +963,32 @@ fn process_write_impl(
...
+impl ScheContext {
+    fn add_statistics(&mut self, cmd_tag: &'static str, stat: &Statistics) {
+        let entry = self.stats.entry(cmd_tag).or_insert_with(Default::default);
+        entry.add_statistics(stat);
+    }
+}
+
+impl ThreadContext for ScheContext {
+    fn on_tick(&mut self) {
+        for (cmd, stat) in self.stats.drain() {
```
Comment (siddontang) [inline src/storage/txn/scheduler.rs]: too many for here, any way to optimize it? /cc @BusyJay

Comment (BusyJay) [reply src/storage/txn/scheduler.rs]: There is not built-in cube support in iterator.

**Comment by siddontang** [comment]
LGTM

Please update the grafana too.

**Pull request merged**